ui/pages/api/stripe/create-checkout-session.ts

```diff
@@ -66,118 +69,165 @@ async function getTaxRates({
 }
 
 export default handler().post(async (req, res) => {
-  if (typeof req.query?.bucketId !== "string") throw new Error("Bad bucketId");
-  if (typeof req.query?.contribution !== "string")
-    throw new Error("Bad contribution");
-  if (typeof req.query?.tipAmount !== "string")
-    throw new Error("Bad tipAmount");
-  const bucketId = req.query?.bucketId;
-  const contribution = Number(req.query?.contribution);
-  const tipAmount = Number(req.query?.tipAmount);
-
-  // throws if not a round member
-  const roundMember = await getRoundMember({
-    bucketId,
-    userId: req.user.id,
-    include: { user: true },
-  });
-
-  const bucket = await prisma.bucket.findUnique({
-    where: { id: bucketId },
-    include: { round: { include: { group: true } } },
-  });
-
-  if (!bucket.directFundingEnabled || !bucket.round.directFundingEnabled) {
-    throw new Error("Direct funding not enabled for this bucket and/or round");
-  }
+  if (req.query?.mode === "paidplan") {
+    if (!req.user) throw new Error("You need to be logged in");
+    if (typeof req.query?.plan !== "string") throw new Error("No plan specified");
+    if (typeof req.query?.groupSlug !== "string") throw new Error("No plan specified");
+    if (typeof req.query?.groupName !== "string") throw new Error("No plan specified");
+
+    //if (typeof req.query?.contribution !== "string")
+
+    const priceId = plans[req.query.plan];
+    if (!priceId) throw new Error("Missing price ID for this plan.");
+
+    const customerMetadata = {
+      customer_email: req.user.email,
+    };
+    const origin = getRequestOrigin(req);
+    try {
+      const session = await stripe.checkout.sessions.create({
+        mode: "subscription",
+        payment_method_types: ["card"],
+        tax_id_collection: {
+          enabled: true,
+        },
```

Comment:
Is this needed?

Reply:
It is to collect VAT ID, I guess many orgs using it will be companies/associations rather than consumers.

Reply:
Ah wait I was thinking this was a donation for some reason. But since they're paying us, shouldn't we add a tax rate like https://github.com/cobudget/cobudget/blob/91ca4288d97de0f333d4853651cc7e843c3727ca/ui/pages/api/stripe/create-checkout-session.ts#L133

Reply:
I'll merge this anyway but add a note in the issue that we might wanna add this before closing